nixos/modules/services/monitoring/aws-xray-daemon.nix

```diff
@@ -0,0 +1,158 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+    cfg = config.services.aws-xray-daemon;
+
+    xrayConfig = {
+        TotalBufferSizeMB = cfg.TotalBufferSizeMB;
+        Concurrency = cfg.Concurrency;
+        Region = cfg.Region;
+        Endpoint = cfg.EndPoint;
+        Socket = {
+          UDPAddress = cfg.Socket.UDPAddress;
+          TCPAddress = cfg.Socket.TCPAddress;
+        };
+        Logging = {
+          LogRotation = cfg.Logging.LogRotation;
+          LogLevel = cfg.Logging.LogLevel;
+          LogPath = cfg.Logging.LogPath;
+        };
+        LocalMode = cfg.LocalMode;
+        ResourceARN = cfg.ResourceARN;
+        RoleARN = cfg.RoleARN;
+        NoVerifySSL = cfg.NoVerifySSL;
+        ProxyAddress = cfg.ProxyAddress;
+        Version = cfg.Version;
+    };
+
+    configFile = pkgs.writeText "xray.yaml" (builtins.toJSON xrayConfig);
```

Comment:
I'm not sure if we should declare every single config attribute in the json config file, maybe it should be just a single string option and we let the user take care of that.